mariadb-plugin/src/main/java/io/cdap/plugin/mariadb/MariadbAction.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.mariadb;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.plugin.db.batch.action.AbstractDBAction;
+import io.cdap.plugin.db.batch.config.DBSpecificQueryConfig;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * MariaDB action that runs MariaDB command.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(MariadbConstants.PLUGIN_NAME)
+@Description("Action that runs a MariaDB command")
+public class MariadbAction extends AbstractDBAction {
+
+  private final MariadbActionConfig mariadbActionConfig;
+
+  /**
+   * This is the constructor for MariadbAction.
+   * @param mariadbActionConfig It takes Mariadb Action config object as the parameter.
+   */
+  public MariadbAction(MariadbActionConfig mariadbActionConfig) {
+    super(mariadbActionConfig, false);
+    this.mariadbActionConfig = mariadbActionConfig;
+  }
+
+  /**
+   * MariaDB Action Config.
+   */
+  public static class MariadbActionConfig extends DBSpecificQueryConfig {
+
+    @Name(MariadbConstants.AUTO_RECONNECT)
```

Comment:
Most of these configs are common to all mariadb plugins. Can you put he common configuration into a class to avoid duplication?

Reply:
TODO

Reply:
The code is modified to specify TODO in comment and reference to JIRA - https://issues.cask.co/browse/PLUGIN-358